app/questionnaire/path_finder.py

```diff
@@ -91,19 +94,50 @@ def _remove_rule_answers(self, goto_rule):
                     self.answer_store.remove(answer_id=condition['id'],
                                              answer_instance=0,)
 
-    def get_routing_path(self, group_id=None, group_instance=0):
+    def get_routing_path(self, group_id, group_instance=0):
         """
         Returns a list of the block ids visited based on answers provided
         :return: List of block location dicts
         """
-        if group_id is None:
-            group_id = SchemaHelper.get_first_group_id(self.survey_json)
+        self.get_full_routing_path()
```

Comment:
feels a bit strange using it this way, would be clearer if the return from this call was assigned to a variable used later on?

Reply:
it essentially is `self.full_routing_path`

Reply:
maybe it should be called `generate_full_routing_path` or something since `get` implies this function returns a value rather than changing the state of the object?

Reply:
`generate_full_routing_path` would be clearer that it's performing a more complex task but the [`build_path`](https://github.com/ONSdigital/eq-survey-runner/blob/261a2de597db5ec44af8ec6b2426c22dd1cabdbc/app/questionnaire/path_finder.py#L28) method ([called](https://github.com/ONSdigital/eq-survey-runner/blob/261a2de597db5ec44af8ec6b2426c22dd1cabdbc/app/questionnaire/path_finder.py#L132) from `get_full_routing_path`) is doing the real heavy lifting and `get_full_routing_path` does then fetch it from the result of that.